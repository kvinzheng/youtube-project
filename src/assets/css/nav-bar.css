#NavBar {
  min-height: 50px;
  background-color: #34373C;
  height: fit-content;
}

#Title {
  margin-left: 2.5%;
  float: left;
}

.MenuItem {
  vertical-align: middle;
  line-height: 50px;
  width: 100%;
  font-family: 'Oswald', sans-serif;
  font-size: 16px;
  display: inline;
  color: #fffcfc;
  font-weight: 700;
  padding-right: 10px;
}

.NavLinks {
  float: right;
  margin-right: 5%;
}

.MenuItem {
  vertical-align: middle;
  line-height: 50px;
  width: 100%;
  font-family: 'Oswald', sans-serif;
  font-size: 16px;
  display: inline;
  color: #fffcfc;
  font-weight: 700;
  padding-right: 10px;
}
