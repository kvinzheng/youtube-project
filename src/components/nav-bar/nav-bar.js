import React from 'react';

const NavBar = () => (
  <div id="NavBar">
    <div id="Title">
      <div className="MenuItem">
        Kevin&apos;s Youtube-Project
      </div>
    </div>
    <div className="NavLinks">
      <a href="https://github.com/kvinzheng/youtube-project">
        <div className="MenuItem">
        GitHub Code
        </div>
      </a>
    </div>
  </div>
);

export default NavBar;
