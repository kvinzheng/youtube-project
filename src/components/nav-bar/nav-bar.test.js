import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NavBar } from './nav-bar';

configure({ adapter: new Adapter() });

const history = {
  push: jest.fn(),
};

describe('NavBar', () => {
  it('should render without any data defined', () => {
    const component = shallow(
      <NavBar
        history={history}
      />);
    expect(component).toMatchSnapshot();
  });

  it('When button is clicked history.push is called', () => {
    const component = mount(
      <NavBar
        history={history}
      />);
    component.find('.home-click').simulate('click');
    expect(history.push).toHaveBeenCalled();
    expect(history.push.mock.calls[0][0]).toBe('/');
  });

  it('Renders with correct classes', () => {
    const component = mount(
      <NavBar
        history={history}
      />);

    expect(component.find('#navbar').length).toBe(1);
    expect(component.find('#title').length).toBe(1);
    expect(component.find('.menu-item').length).toBe(3);
    expect(component.find('.home').length).toBe(1);
    expect(component.find('.home-click').length).toBe(1);
    expect(component.find('.nav-links').length).toBe(1);
  });
});
