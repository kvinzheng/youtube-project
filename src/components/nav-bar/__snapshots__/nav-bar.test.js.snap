// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render without any data defined 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="title"
>
          <div
                    className="menu-item"
          >
                    Kevin's Youtube-Project
          </div>
</div>,
        <div
          className="home"
>
          <a
                    className="home-click"
                    onClick={[Function]}
          >
                    <div
                              className="menu-item"
                    >
                              Home Page
                    </div>
          </a>
</div>,
        <div
          className="nav-links"
>
          <a
                    href="https://github.com/kvinzheng/youtube-project"
          >
                    <div
                              className="menu-item"
                    >
                              GitHub Code
                    </div>
          </a>
</div>,
      ],
      "id": "navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="menu-item"
>
            Kevin's Youtube-Project
</div>,
          "id": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Kevin's Youtube-Project",
            "className": "menu-item",
          },
          "ref": null,
          "rendered": "Kevin's Youtube-Project",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="home-click"
            onClick={[Function]}
>
            <div
                        className="menu-item"
            >
                        Home Page
            </div>
</a>,
          "className": "home",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="menu-item"
>
              Home Page
</div>,
            "className": "home-click",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Home Page",
              "className": "menu-item",
            },
            "ref": null,
            "rendered": "Home Page",
            "type": "div",
          },
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://github.com/kvinzheng/youtube-project"
>
            <div
                        className="menu-item"
            >
                        GitHub Code
            </div>
</a>,
          "className": "nav-links",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="menu-item"
>
              GitHub Code
</div>,
            "href": "https://github.com/kvinzheng/youtube-project",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "GitHub Code",
              "className": "menu-item",
            },
            "ref": null,
            "rendered": "GitHub Code",
            "type": "div",
          },
          "type": "a",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="title"
>
            <div
                        className="menu-item"
            >
                        Kevin's Youtube-Project
            </div>
</div>,
          <div
            className="home"
>
            <a
                        className="home-click"
                        onClick={[Function]}
            >
                        <div
                                    className="menu-item"
                        >
                                    Home Page
                        </div>
            </a>
</div>,
          <div
            className="nav-links"
>
            <a
                        href="https://github.com/kvinzheng/youtube-project"
            >
                        <div
                                    className="menu-item"
                        >
                                    GitHub Code
                        </div>
            </a>
</div>,
        ],
        "id": "navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="menu-item"
>
              Kevin's Youtube-Project
</div>,
            "id": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Kevin's Youtube-Project",
              "className": "menu-item",
            },
            "ref": null,
            "rendered": "Kevin's Youtube-Project",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="home-click"
              onClick={[Function]}
>
              <div
                            className="menu-item"
              >
                            Home Page
              </div>
</a>,
            "className": "home",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="menu-item"
>
                Home Page
</div>,
              "className": "home-click",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Home Page",
                "className": "menu-item",
              },
              "ref": null,
              "rendered": "Home Page",
              "type": "div",
            },
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://github.com/kvinzheng/youtube-project"
>
              <div
                            className="menu-item"
              >
                            GitHub Code
              </div>
</a>,
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="menu-item"
>
                GitHub Code
</div>,
              "href": "https://github.com/kvinzheng/youtube-project",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "GitHub Code",
                "className": "menu-item",
              },
              "ref": null,
              "rendered": "GitHub Code",
              "type": "div",
            },
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
